### PR TITLE
[TACHYON-562] Improved condition handling in MasterClient

### PR DESCRIPTION
**Pull request by yupeng9**
https://tachyon.atlassian.net/browse/TACHYON-562
Replaced the while loop with if condition

### CODE REVIEW

**Comment by calvinjia** [comment]
Jenkins, add to whitelist.

**Comment by calvinjia** [comment]
Jenkins, test this please.

File: common/src/main/java/tachyon/master/MasterClient.java [hunk 1]
```diff
@@ -303,12 +303,12 @@ private synchronized InetSocketAddress getMasterAddress() {
    * @throws IOException
    */
   public synchronized long getUserId() throws IOException {
-    while (!mIsClosed) {
+    if (mIsClosed) {
+      return -1;
+    } else {
```
Comment (calvinjia) [inline common/src/main/java/tachyon/master/MasterClient.java]: I think we try to not use `else` after an `if` that does a `return`. @apc999

Comment (yupeng9) [inline common/src/main/java/tachyon/master/MasterClient.java]: yep, I'm fine with this convention, given it's more concise

**Comment by calvinjia** [comment]
+1 LGTM

**Comment by apc999** [comment]
+1 LGTM

**Comment by haoyuan** [comment]
Why change the while loop to be a if condition? Thanks.

**Comment by apc999** [comment]
@haoyuan  there is a return in while loop, thus while is never really looping anyway

**Pull request merged**